voltdbclient/url.go

```diff
@@ -0,0 +1,33 @@
+package voltdbclient
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+)
+
+var errBadURL = errors.New("voltdb: bad connection url")
+
+func parseURL(src string) (*url.URL, error) {
+	u, err := url.Parse(src)
+	if err != nil {
+		return nil, err
+	}
+	if u.User == nil {
+		u.User = &url.Userinfo{}
+	}
+	if u.Scheme != "voltdb" {
```

Comment:
We can do that, but I was afraid to break backward compatibility. for instance if you connect with the following url

`localhost:21212`

It will be parsed to

```
&url.URL{
    Scheme:     "localhost",
    Opaque:     "21212",
    User:       (*url.Userinfo)(nil),
    Host:       "",
    Path:       "",
    RawPath:    "",
    ForceQuery: false,
    RawQuery:   "",
    Fragment:   "",
}
```
I thought it is a good thing to allow this to work. But, if we need to be safe I agree with you. We can enforce the scheme to be `voltdb` but again I will need to touch a lot of other places to refactor the connection scheme.